File: BeiBei2DXLua/frameworks/cocos2d-x/common/Android/CXAvos.cpp
//
//  CXAvos.cpp
//  BeiBei2DXLua
//
//  Created by Bemoo on 10/8/14.
//
//

#include "CXAvos.h"
#include "CCLuaEngine.h"
#include "platform/android/jni/JniHelper.h"

#define JAVA_PKG "c/bb/dc/BBNDK"

using namespace cocos2d;

void CXAvos::init() {
}

void CXAvos::downloadFile(const char* objectId, const char* savepath, CXLUAFUNC nHandler) {
    mLuaHandlerId_dl = nHandler;

    cocos2d::JniMethodInfo t;
    if (cocos2d::JniHelper::getStaticMethodInfo(t, JAVA_PKG, "downloadFile", "(Ljava/lang/String;Ljava/lang/String;)V")) {
        jstring stringArg_objectId = t.env->NewStringUTF(objectId);
        jstring stringArg_savepath = t.env->NewStringUTF(savepath);

        t.env->CallStaticVoidMethod(t.classID, t.methodID, stringArg_objectId, stringArg_savepath);

        t.env->DeleteLocalRef(stringArg_objectId);
        t.env->DeleteLocalRef(stringArg_savepath);
        t.env->DeleteLocalRef(t.classID);
    }
}

void CXAvos::downloadConfigFiles(const char* objectIds, const char* path) {
    cocos2d::JniMethodInfo t;
    if (cocos2d::JniHelper::getStaticMethodInfo(t, JAVA_PKG, "downloadConfigFiles", "(Ljava/lang/String;Ljava/lang/String;)V")) {
        jstring stringArg_objectIds = t.env->NewStringUTF(objectIds);
        jstring stringArg_path = t.env->NewStringUTF(path);

        t.env->CallStaticVoidMethod(t.classID, t.methodID, stringArg_objectIds, stringArg_path);

        t.env->DeleteLocalRef(stringArg_objectIds);
        t.env->DeleteLocalRef(stringArg_path);
        t.env->DeleteLocalRef(t.classID);
    }
}

void CXAvos::downloadWordSoundFiles(const char* prefix, const char* wordsList, const char* subfix, const char* path) {
    cocos2d::JniMethodInfo t;
    if (cocos2d::JniHelper::getStaticMethodInfo(t, JAVA_PKG, "downloadWordSoundFiles", "(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)V")) {
        jstring stringArg_prefix = t.env->NewStringUTF(prefix);
        jstring stringArg_wordsList = t.env->NewStringUTF(wordsList);
        jstring stringArg_subfix = t.env->NewStringUTF(subfix);
        jstring stringArg_path = t.env->NewStringUTF(path);

        t.env->CallStaticVoidMethod(t.classID, t.methodID, stringArg_prefix, stringArg_wordsList, stringArg_subfix, stringArg_path);

        t.env->DeleteLocalRef(stringArg_prefix);
        t.env->DeleteLocalRef(stringArg_wordsList);
        t.env->DeleteLocalRef(stringArg_subfix);
        t.env->DeleteLocalRef(stringArg_path);
        t.env->DeleteLocalRef(t.classID);
    }
}

void CXAvos::signUp(const char* username, const char* password, CXLUAFUNC nHandler) {
    mLuaHandlerId_signUp = nHandler;
    cocos2d::JniMethodInfo t;
    if (cocos2d::JniHelper::getStaticMethodInfo(t, JAVA_PKG, "signUp", "(Ljava/lang/String;Ljava/lang/String;)V")) {
        jstring stringArg_username = t.env->NewStringUTF(username);
        jstring stringArg_password = t.env->NewStringUTF(password);

        t.env->CallStaticVoidMethod(t.classID, t.methodID, stringArg_username, stringArg_password);

        t.env->DeleteLocalRef(stringArg_username);
        t.env->DeleteLocalRef(stringArg_password);
        t.env->DeleteLocalRef(t.classID);
    }
}

void CXAvos::logIn(const char* username, const char* password, CXLUAFUNC nHandler) {
    mLuaHandlerId_logIn = nHandler;
    cocos2d::JniMethodInfo t;
    if (cocos2d::JniHelper::getStaticMethodInfo(t, JAVA_PKG, "logIn", "(Ljava/lang/String;Ljava/lang/String;)V")) {
        jstring stringArg_username = t.env->NewStringUTF(username);
        jstring stringArg_password = t.env->NewStringUTF(password);

        t.env->CallStaticVoidMethod(t.classID, t.methodID, stringArg_username, stringArg_password);

        t.env->DeleteLocalRef(stringArg_username);
        t.env->DeleteLocalRef(stringArg_password);
        t.env->DeleteLocalRef(t.classID);
    }
}

void CXAvos::initTencentQQ(const char* appId, const char* appKey) {
    cocos2d::JniMethodInfo t;
    if (cocos2d::JniHelper::getStaticMethodInfo(t, JAVA_PKG, "initTencentQQ", "(Ljava/lang/String;Ljava/lang/String;)V")) {
        jstring jappId = t.env->NewStringUTF(appId);
        jstring jappKey = t.env->NewStringUTF(appKey);

        t.env->CallStaticVoidMethod(t.classID, t.methodID, jappId, jappKey);

        t.env->DeleteLocalRef(jappId);
        t.env->DeleteLocalRef(jappKey);
        t.env->DeleteLocalRef(t.classID);
    }
}

void CXAvos::logInByQQ(CXLUAFUNC nHandler) {
    mLuaHandlerId_logInByQQ = nHandler;
    cocos2d::JniMethodInfo t;
    if (cocos2d::JniHelper::getStaticMethodInfo(t, JAVA_PKG, "logInByQQ", "()V")) {
        t.env->CallStaticVoidMethod(t.classID, t.methodID);
        t.env->DeleteLocalRef(t.classID);
    }
}

void CXAvos::logInByQQAuthData(const char* openid, const char* access_token, const char* expires_in, CXLUAFUNC nHandler) {
    mLuaHandlerId_logInByQQ = nHandler;
    cocos2d::JniMethodInfo t;
    if (cocos2d::JniHelper::getStaticMethodInfo(t, JAVA_PKG, "logInByQQAuthData", "(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)V")) {
        jstring stringArg_openid = t.env->NewStringUTF(openid);
        jstring stringArg_access_token = t.env->NewStringUTF(access_token);
        jstring stringArg_expires_in = t.env->NewStringUTF(expires_in);

        t.env->CallStaticVoidMethod(t.classID, t.methodID, stringArg_openid, stringArg_access_token, stringArg_expires_in);

        t.env->DeleteLocalRef(stringArg_openid);
        t.env->DeleteLocalRef(stringArg_access_token);
        t.env->DeleteLocalRef(stringArg_expires_in);
        t.env->DeleteLocalRef(t.classID);
    }
}

void CXAvos::logOut() {
    cocos2d::JniMethodInfo t;
    if (cocos2d::JniHelper::getStaticMethodInfo(t, JAVA_PKG, "logOut", "()V")) {
        t.env->CallStaticVoidMethod(t.classID, t.methodID);
        t.env->DeleteLocalRef(t.classID);
    }
}

File: BeiBei2DXLua/frameworks/cocos2d-x/cocos/scripting/lua-bindings/auto/api/CXAvos.lua

--------------------------------
-- @module CXAvos
-- @extend Ref
-- @parent_module cx

--------------------------------
-- 
-- @function [parent=#CXAvos] invokeLuaCallbackFunction_su 
-- @param self
-- @param #char objectjson
-- @param #char error
-- @param #int errorcode
        
--------------------------------
-- 请求短信验证码
-- @function [parent=#CXAvos] requestSMSCode 
-- @param self
-- @param #char phoneNumber
        
--------------------------------
-- 验证短信验证码
-- @function [parent=#CXAvos] verifySMSCode 
-- @param self
-- @param #char phoneNumber
-- @param #char smsCode
-- @param #int mHandler
        
--------------------------------
-- 
-- @function [parent=#CXAvos] downloadFile 
-- @param self
-- @param #char objectId
-- @param #char savepath
-- @param #int nHandler
        
--------------------------------
-- 
-- @function [parent=#CXAvos] invokeLuaCallbackFunction_logInByQQ 
-- @param self
-- @param #char objectjson
-- @param #char qqjson
-- @param #char authjson
-- @param #char error
-- @param #int errorcode
        
--------------------------------
-- 
-- @function [parent=#CXAvos] downloadWordSoundFiles 
-- @param self
-- @param #char prefix
-- @param #char wordsList
-- @param #char subfix
-- @param #char path
        
--------------------------------
-- 
-- @function [parent=#CXAvos] logIn 
-- @param self
-- @param #char username
-- @param #char password
-- @param #int nHandler
        
--------------------------------
-- 
-- @function [parent=#CXAvos] invokeLuaCallbackFunction_li 
-- @param self
-- @param #char objectjson
-- @param #char error
-- @param #int errorcode
        
--------------------------------
-- 使用SMS Code登陆
-- @function [parent=#CXAvos] loginWithSMS 
-- @param self
-- @param #char phoneNumber
-- @param #char smsCode
-- @param #int mHandler
        
--------------------------------
-- 
-- @function [parent=#CXAvos] logInByQQAuthData 
-- @param self
-- @param #char openid
-- @param #char access_token
-- @param #char expires_in
-- @param #int nHandler
        
--------------------------------
-- 
-- @function [parent=#CXAvos] invokeLuaCallbackFunction_dl 
-- @param self
-- @param #char objectId
-- @param #char filename
-- @param #char error
-- @param #bool isSaved
        
--------------------------------
-- 验证之后的回调
-- @function [parent=#CXAvos] invokeLuaCallBackFunction_vp 
-- @param self
-- @param #char error
-- @param #int errorCode
        
--------------------------------
-- 
-- @function [parent=#CXAvos] logInByQQ 
-- @param self
-- @param #int nHandler
        
--------------------------------
-- 请求验证手机号码 服务器会给手机发送验证码
-- @function [parent=#CXAvos] requestVerifyPhoneNumber 
-- @param self
-- @param #char phoneNumber
        
--------------------------------
-- 验证短信验证码 --这干码 跟上边那个码不一样,只是用来验证手机号有效性的,上边是跟账号无关的验证
-- @function [parent=#CXAvos] verifyPhoneNumber 
-- @param self
-- @param #char phoneNumber
-- @param #char smsCode
-- @param #int mHandler
        
--------------------------------
-- 修改密码
-- @function [parent=#CXAvos] changePwd 
-- @param self
-- @param #char username
-- @param #char oldPwd
-- @param #char newPwd
-- @param #int nHandler
        
--------------------------------
-- 
-- @function [parent=#CXAvos] logInByPhoneNumber 
-- @param self
-- @param #char PhoneNumber
-- @param #char password
-- @param #int nHandler
        
--------------------------------
-- 
-- @function [parent=#CXAvos] downloadConfigFiles 
-- @param self
-- @param #char objectIds
-- @param #char path
        
--------------------------------
-- 
-- @function [parent=#CXAvos] initTencentQQ 
-- @param self
-- @param #char appId
-- @param #char appKey
        
--------------------------------
-- 用验证码登陆回调
-- @function [parent=#CXAvos] invokeLuaCallBackFunction_ls 
-- @param self
-- @param #char objectjson
-- @param #char error
-- @param #int errorcode
        
--------------------------------
-- 
-- @function [parent=#CXAvos] signUp 
-- @param self
-- @param #char username
-- @param #char password
-- @param #int nHandler
        
--------------------------------
-- 修改密码完成的回调
-- @function [parent=#CXAvos] invokeLuaCallbackFunction_cp 
-- @param self
-- @param #char error
-- @param #int errorcode
        
--------------------------------
-- 
-- @function [parent=#CXAvos] logOut 
-- @param self
        
--------------------------------
-- 获取手机登陆的验证码
-- @function [parent=#CXAvos] requestLoginSMS 
-- @param self
-- @param #char phoneNumber
        
--------------------------------
-- 验证之后的回调
-- @function [parent=#CXAvos] invokeLuaCallBackFunction_vc 
-- @param self
-- @param #char error
-- @param #int errorCode
        
--------------------------------
-- 
-- @function [parent=#CXAvos] getInstance 
-- @param self
-- @return CXAvos#CXAvos ret (return value: CXAvos)
        
return nil
